setup/advanced/kubernetes-sidecar.md

```diff
@@ -0,0 +1,119 @@
+# Kubernetes Sidecar Container Setup
+
+To get started with a Kubernetes sidecar container setup, simply adapt the following deployment yaml to suit your needs. Be sure to create a persistent volume and persistent volume claim for Gluetun to be able to store data!
+
+```yaml
+---
+ apiVersion: apps/v1
+ kind: Deployment
+ metadata:
+   name: gluetun-deployment
+   labels:
+     app: gluetun
+ spec:
+   replicas: 1
+   strategy:
+       type: RollingUpdate
+       rollingUpdate:
+          maxSurge: 1
+   selector:
+      matchLabels:
+        app: gluetun
+   template:
+     metadata:
+       labels:
+         app: gluetun
+     spec:
+      containers:
+         - name: gluetun
+           image: ghcr.io/qdm12/gluetun # Optionally you can use the "qmcgaw/gluetun" image as well as specify what version of Gluetun you desire
+           imagePullPolicy: Always
+           securityContext:
+             capabilities:
+               add: ["NET_ADMIN"]
+           # You can optionally specify ports that are used by your containers, but it is not strictly necessary
+           # ports:
+             # - containerPort: 8080
+
+           # Adapt the following environment variables to suit your needs and VPN provider
+           env:
+           - name: TZ
+             value: "Europe/London"
+
+           - name: VPN_SERVICE_PROVIDER
+             value: "ivpn"
+
+           - name: OPENVPN_USER
+             value: ""
+
+           - name: OPENVPN_PASSWORD
+             value: ""
+
+           # If you having connection issues, try enabling these variables to help diagnose it.
+           # - name: FIREWALL_DEBUG
+           #   value: "on"
+           # - name: FIREWALL_INPUT_PORTS
+           #   value: "8080"
+
+           volumeMounts:
+             - name: gluetun-config
+               mountPath: /gluetun
+
+
+           # -- Connecting Other Containers --
+         # Define other containers that you want to connected to a VPN.
+         # When using Gluetun in a sidecar configuration, all other containers will use Gluetun's VPN connection.
+         # For testing purposes, you can `kubectl exec` into this curl container and run `curl https://ipinfo.io` to test your connection!
+
+         - name: curl-container
+           image: quay.io/curl/curl:latest
+           command: ["sleep", "infinity"]
```

Comment:
Changing this a bit:

- Testing it already documented in https://github.com/qdm12/gluetun-wiki/blob/main/setup/test-your-setup.md and I think K8s+Gluetun users would be aware of this. You could however add after this yml the equivalent `kubectl exec gluetun ...` to test the vpn connection.
- Leaving a simplified `alpine` container as a commented example, it's better to not have an unneeded curl container the user forgets to comment out.

```suggestion
         # -- Connecting Other Containers --
         # Define other containers you want connected to the VPN through Gluetun.
         # An example of container is commented below.

         # - name: gluetun-alpine
         #   image: alpine
          #  command: ["sleep", "infinity"]
```

Reply:
it will help to document how to actually test the connection from alpine's shell:
`echo "wget https://ipinfo.io" | kubectl exec -i -t pod-name --container gluetun-alpine -- /bin/ash`